lib/veritech-server/src/server.rs

```diff
@@ -213,6 +243,73 @@ impl Server {
             cyclone_client_execution_timeout,
             nats,
             kill_senders,
+            None,
+        );
+
+        let app = ServiceBuilder::new()
+            .layer(
+                MatchedSubjectLayer::new()
+                    .for_subject(VeritechForSubject::with_prefix(prefix.as_deref())),
+            )
+            .layer(
+                TraceLayer::new()
+                    .make_span_with(
+                        telemetry_nats::NatsMakeSpan::builder(connection_metadata).build(),
+                    )
+                    .on_response(telemetry_nats::NatsOnResponse::new()),
+            )
+            .layer(AckLayer::new())
+            .service(handlers::process_request.with_state(state))
+            .map_response(Response::into_response);
+
+        // Reset metrics before creating the naxum server.
+        metric!(counter.veritech.pool_exhausted = 0);
```

Comment:
Ah I moved this above and missed a spot to delete it here. Will do later. It's harmless but shouldn't be here since it's now above.